### PR TITLE
Add note about keeping the middleware pipeline running for a WebSocket

### PR DESCRIPTION
**Pull request by anurse**
Related to https://github.com/aspnet/AspNetCore/issues/2854

I think I got the API doc xrefs right but we'll see :).

[Internal Review Version](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/websockets?view=aspnetcore-1.1&branch=pr-en-us-12378)

### CODE REVIEW

File: aspnetcore/fundamentals/websockets.md [hunk 1]
```diff
@@ -117,6 +117,15 @@ The following example is from later in the `Configure` method:
...
 A WebSocket request could come in on any URL, but this sample code only accepts requests for `/ws`.
 
+When using a WebSocket, you **must** keep the middleware pipeline running for the duration of the connection. If you attempt to send or receive a WebSocket message after the middleware pipeline ends, you may get an exception like the following:
+
+```
+System.Net.WebSockets.WebSocketException (0x80004005): The remote party closed the WebSocket connection without completing the close handshake. ---> System.ObjectDisposedException: Cannot write to the response body, the response has completed.
+Object name: 'HttpResponseStream'.
+```
+
+If you are using a background service to write data to a WebSocket, make sure you keep the middleware pipeline running. You can do this by using a <xref:System.Threading.Tasks.TaskCompletionSource%601>. Pass the TaskCompletionSource to your background service and have it call <xref:System.Threading.Tasks.TaskCompletionSource%601.TrySetResult%2A> when you finish with the WebSocket, then `await` the <xref:System.Threading.Tasks.TaskCompletionSource%601.Task> property during the request.
```
Comment (scottaddie) [inline aspnetcore/fundamentals/websockets.md]:
```suggestion
If you're using a background service to write data to a WebSocket, make sure you keep the middleware pipeline running. Do this by using a <xref:System.Threading.Tasks.TaskCompletionSource%601>. Pass the `TaskCompletionSource` to your background service and have it call <xref:System.Threading.Tasks.TaskCompletionSource%601.TrySetResult%2A> when you finish with the WebSocket. Then `await` the <xref:System.Threading.Tasks.TaskCompletionSource%601.Task> property during the request.
```

Comment (anurse) [reply aspnetcore/fundamentals/websockets.md]: FOILED AGAIN!

Comment (anurse) [reply aspnetcore/fundamentals/websockets.md]: Contractions... my old adversary.

sideshow bob from the simpsons stepping on a lot of rakes

**Comment by tdykstra** [comment]
Fixes #12629

**Pull request merged**